comp/core/tagger/impl-remote/remote.go

```diff
@@ -250,6 +256,68 @@ func (t *remoteTagger) LegacyTag(entity string, cardinality types.TagCardinality
 	return t.Tag(entityID, cardinality)
 }
 
+// GenerateContainerIDFromExternalData returns a container ID for the given external data.
+func (t *remoteTagger) GenerateContainerIDFromExternalData(externalData origindetection.ExternalData) (string, error) {
+	expBackoff := backoff.NewExponentialBackOff()
+	expBackoff.InitialInterval = 500 * time.Millisecond
+	expBackoff.MaxInterval = 1 * time.Second
+	expBackoff.MaxElapsedTime = 15 * time.Second
+
+	var containerID string
+
+	err := backoff.Retry(func() error {
+		select {
+		case <-t.telemetryTicker.C:
+			t.store.collectTelemetry()
```

Comment:
We shouldn't do this here. The telemetry should be collected only by the `run()` function.